perception/autoware_tensorrt_rtmdet/include/tensorrt_rtmdet/calibrator.hpp

```diff
@@ -0,0 +1,502 @@
+    double quantile = 0.999999, double cutoff = 0.999999)
+  : stream_(stream),
+    calibration_cache_file_(calibration_cache_file),
+    histogram_cache_file_(histogram_cache_file),
+    read_cache_(read_cache)
+  {
+    auto d = stream_.getInputDims();
+    input_count_ = stream_.getBatchSize() * d.d[1] * d.d[2] * d.d[3];
+    CHECK_CUDA_ERROR(cudaMalloc(&device_input_, input_count_ * sizeof(float)));
+    m_std = std;
+    m_mean = mean;
+    m_quantile = quantile;
+    m_cutoff = cutoff;
+    auto algType = getAlgorithm();
+    switch (algType) {
+      case (nvinfer1::CalibrationAlgoType::kLEGACY_CALIBRATION):
+        std::cout << "CalibrationAlgoType : kLEGACY_CALIBRATION" << std::endl;
+        break;
+      case (nvinfer1::CalibrationAlgoType::kENTROPY_CALIBRATION):
+        std::cout << "CalibrationAlgoType : kENTROPY_CALIBRATION" << std::endl;
+        break;
+      case (nvinfer1::CalibrationAlgoType::kENTROPY_CALIBRATION_2):
+        std::cout << "CalibrationAlgoType : kENTROPY_CALIBRATION_2" << std::endl;
+        break;
+      case (nvinfer1::CalibrationAlgoType::kMINMAX_CALIBRATION):
+        std::cout << "CalibrationAlgoType : kMINMAX_CALIBRATION" << std::endl;
+        break;
+      default:
+        std::cout << "No CalibrationAlgType" << std::endl;
+        break;
+    }
+  }
+  int getBatchSize() const noexcept override { return stream_.getBatchSize(); }
+
+  virtual ~Int8LegacyCalibrator() { CHECK_CUDA_ERROR(cudaFree(device_input_)); }
+
+  bool getBatch(void * bindings[], const char * names[], int nb_bindings) noexcept override
+  {
+    (void)names;
+    (void)nb_bindings;
+
+    if (!stream_.next(m_mean, m_std)) {
+      return false;
+    }
+    try {
+      CHECK_CUDA_ERROR(cudaMemcpy(
+        device_input_, stream_.getBatch(), input_count_ * sizeof(float), cudaMemcpyHostToDevice));
+    } catch (const std::exception & e) {
+      // Do nothing
+    }
+    bindings[0] = device_input_;
+    return true;
+  }
+
+  const void * readCalibrationCache(size_t & length) noexcept override
+  {
+    calib_cache_.clear();
+    std::ifstream input(calibration_cache_file_, std::ios::binary);
+    input >> std::noskipws;
+    if (read_cache_ && input.good()) {
+      std::copy(
+        std::istream_iterator<char>(input), std::istream_iterator<char>(),
+        std::back_inserter(calib_cache_));
+    }
+
+    length = calib_cache_.size();
+    if (length) {
+      std::cout << "Using cached calibration table to build the engine" << std::endl;
+    } else {
+      std::cout << "New calibration table will be created to build the engine" << std::endl;
+    }
+    return length ? &calib_cache_[0] : nullptr;
+  }
+
+  void writeCalibrationCache(const void * cache, size_t length) noexcept override
+  {
+    std::ofstream output(calibration_cache_file_, std::ios::binary);
+    output.write(reinterpret_cast<const char *>(cache), length);
+  }
+
+  double getQuantile() const noexcept
+  {
+    printf("Quantile %f\n", m_quantile);
```

Comment:
Thank you for your PR!

Please use `RCLCPP_INFO` or `RCLCPP_INFO_THROTTLE`, instead of `printf` or `std::cout`

Thank you for your contribution!

Reply:
Changed all of the `printf` and `std::cout` functions with `RCLCPP_INFO`.

related commit: https://github.com/autowarefoundation/autoware.universe/pull/8165/commits/7abd6caf6c99d22274d66f7c42b5f61b30b4f99a